### ISSUE TITLE
Error on build: double free or corruption (out) SIGABRT: abort PC=0x7f7464f01e97 m=0 sigcode=18446744073709551610 signal arrived during cgo execution

### ISSUE THREAD

**Issue by Sarke** [moby/moby#37916]
**Description**

See error below.

**Steps to reproduce the issue:**
1. `sudo docker build -t site-watcher .`

**Describe the results you received:**
```
double free or corruption (out)
SIGABRT: abort
PC=0x7f3e05499e97 m=0 sigcode=18446744073709551610
signal arrived during cgo execution

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x4afd50, 0xc42004fcc0, 0xc42004fce8)
	/usr/lib/go-1.8/src/runtime/cgocall.go:131 +0xe2 fp=0xc42004fc90 sp=0xc42004fc50
github.com/docker/docker-credential-helpers/secretservice._Cfunc_free(0xc6a920)
	github.com/docker/docker-credential-helpers/secretservice/_obj/_cgo_gotypes.go:111 +0x41 fp=0xc42004fcc0 sp=0xc42004fc90
github.com/docker/docker-credential-helpers/secretservice.Secretservice.List.func5(0xc6a920)
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/obj-x86_64-linux-gnu/src/github.com/docker/docker-credential-helpers/secretservice/secretservice_linux.go:96 +0x60 fp=0xc42004fcf8 sp=0xc42004fcc0
github.com/docker/docker-credential-helpers/secretservice.Secretservice.List(0x0, 0x756060, 0xc4200123c0)
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/obj-x86_64-linux-gnu/src/github.com/docker/docker-credential-helpers/secretservice/secretservice_linux.go:97 +0x217 fp=0xc42004fda0 sp=0xc42004fcf8
github.com/docker/docker-credential-helpers/secretservice.(*Secretservice).List(0x77e548, 0xc42004fe88, 0x410022, 0xc420012320)
	<autogenerated>:4 +0x46 fp=0xc42004fde0 sp=0xc42004fda0
github.com/docker/docker-credential-helpers/credentials.List(0x756ba0, 0x77e548, 0x7560e0, 0xc42000e018, 0x0, 0x10)
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/obj-x86_64-linux-gnu/src/github.com/docker/docker-credential-helpers/credentials/credentials.go:145 +0x3e fp=0xc42004fe68 sp=0xc42004fde0
github.com/docker/docker-credential-helpers/credentials.HandleCommand(0x756ba0, 0x77e548, 0x7ffd71137855, 0x4, 0x7560a0, 0xc42000e010, 0x7560e0, 0xc42000e018, 0x40e398, 0x4d35c0)
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/obj-x86_64-linux-gnu/src/github.com/docker/docker-credential-helpers/credentials/credentials.go:60 +0x16d fp=0xc42004fed8 sp=0xc42004fe68
github.com/docker/docker-credential-helpers/credentials.Serve(0x756ba0, 0x77e548)
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/obj-x86_64-linux-gnu/src/github.com/docker/docker-credential-helpers/credentials/credentials.go:41 +0x1cb fp=0xc42004ff58 sp=0xc42004fed8
main.main()
	/build/golang-github-docker-docker-credential-helpers-cMhSy1/golang-github-docker-docker-credential-helpers-0.5.0/secretservice/cmd/main_linux.go:9 +0x4f fp=0xc42004ff88 sp=0xc42004ff58
runtime.main()
	/usr/lib/go-1.8/src/runtime/proc.go:185 +0x20a fp=0xc42004ffe0 sp=0xc42004ff88
runtime.goexit()
	/usr/lib/go-1.8/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc42004ffe8 sp=0xc42004ffe0

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/go-1.8/src/runtime/asm_amd64.s:2197 +0x1

rax    0x0
rbx    0x7ffd711368c0
rcx    0x7f3e05499e97
rdx    0x0
rdi    0x2
rsi    0x7ffd71136650
rbp    0x7ffd711369c0
rsp    0x7ffd71136650
r8     0x0
r9     0x7ffd71136650
r10    0x8
r11    0x246
r12    0x7ffd711368c0
r13    0x1000
r14    0x0
r15    0x30
rip    0x7f3e05499e97
rflags 0x246
cs     0x33
fs     0x0
gs     0x0
Sending build context to Docker daemon  18.43kB
Step 1/8 : FROM node:10-slim
 ---> c33688dea543
```
Rest of build proceeds normally.

**Describe the results you expected:**
Not seeing the error info.

**Additional information you deem important (e.g. issue happens only occasionally):**
It takes about 20 seconds before the output above appears.  It happens every time, and I've tried with different projects.

**Output of `docker version`:**

```
Docker version 18.06.1-ce, build e68fc7a
```
same thing happened with an earlier version
```
Docker version 17.12.1-ce, build 7390fc6
```

**Output of `docker info`:**

```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 15
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.15.0-33-generic
Operating System: Ubuntu 18.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.35GiB
Name: peter-office
ID: Y2JB:LSMD:DZPK:E5UG:T6ZW:4SD6:FYVT:ZFDO:KKEV:3ADM:2MHP:KNGX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

**Comment by Sarke** [comment]
This appears to be the same issue: https://github.com/docker/docker-credential-helpers/issues/103

**Comment by Sarke** [comment]
@aidanh010 found the problem:
https://github.com/docker/docker-credential-helpers/issues/103#issuecomment-421822269

```
$ sudo apt remove golang-docker-credential-helpers
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  docker-compose golang-docker-credential-helpers python-docker
  python-dockerpycreds
0 upgraded, 0 newly installed, 4 to remove and 0 not upgraded.
After this operation, 2,438 kB disk space will be freed.
Do you want to continue? [Y/n] 
(Reading database ... 276129 files and directories currently installed.)
Removing docker-compose (1.17.1-2) ...
Removing python-docker (2.5.1-1) ...
Removing python-dockerpycreds (0.2.1-1) ...
Removing golang-docker-credential-helpers (0.5.0-2) ...
Processing triggers for man-db (2.8.3-2) ...
```
Build runs fine after that.

Shame it removes `docker-compose` as well.

**Comment by thaJeztah** [comment]
@Sarke did you install your version of docker from the official docker package repositories, or from the distro (Ubuntu) repository? The `golang-docker-credential-helpers` package is not part of the official docker packages, so wondering if your docker version is also built by the distro packagers, and something is funky with those builds

**Comment by Sarke** [comment]
@thaJeztah for `docker` I used the official `https://download.docker.com/linux/ubuntu bionic` repo, as per the documentation.

However, for `docker-compose`, which depends on `golang-docker-credential-helpers`, it comes from the official Ubuntu repo.

`$ apt-cache showpkg golang-docker-credential-helpers`
```
Package: golang-docker-credential-helpers
Versions: 
0.5.0-2 (/var/lib/apt/lists/mirror.it.ubc.ca_ubuntu_dists_bionic_universe_binary-amd64_Packages) (/var/lib/dpkg/status)
 Description Language: 
                 File: /var/lib/apt/lists/mirror.it.ubc.ca_ubuntu_dists_bionic_universe_binary-amd64_Packages
                  MD5: 644749d414f966cf290307b53116aed8
 Description Language: 
                 File: /var/lib/apt/lists/mirror.it.ubc.ca_ubuntu_dists_bionic_universe_binary-i386_Packages
                  MD5: 644749d414f966cf290307b53116aed8
 Description Language: en
                 File: /var/lib/apt/lists/mirror.it.ubc.ca_ubuntu_dists_bionic_universe_i18n_Translation-en
                  MD5: 644749d414f966cf290307b53116aed8


Reverse Depends: 
  python-dockerpycreds,golang-docker-credential-helpers
  golang-docker-credential-helpers:i386,golang-docker-credential-helpers
  python3-dockerpycreds,golang-docker-credential-helpers
Dependencies: 
0.5.0-2 - libc6 (2 2.14) libglib2.0-0 (2 2.28.0) libsecret-1-0 (2 0.7) golang-docker-credential-helpers:i386 (32 (null)) 
Provides: 
0.5.0-2 - 
Reverse Provides: 
```

`$ dpkg -s golang-docker-credential-helpers`
```
Package: golang-docker-credential-helpers
Status: install ok installed
Priority: extra
Section: devel
Installed-Size: 1435
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Source: golang-github-docker-docker-credential-helpers
Version: 0.5.0-2
Depends: libc6 (>= 2.14), libglib2.0-0 (>= 2.28.0), libsecret-1-0 (>= 0.7)
Description: Use native stores to safeguard Docker credentials
 The docker-credential-helpers package is a suite of programs to use
 the built-in OS native stores to keep Docker credentials safe.
 .
 This package contains the command-line executables.
Original-Maintainer: Debian Go Packaging Team <pkg-go-maintainers@lists.alioth.debian.org>
Built-Using: golang-1.8 (= 1.8.1-1ubuntu3)
Homepage: https://github.com/docker/docker-credential-helpers
```

**Comment by aidanh010** [comment]
Filed a bug with Ubuntu a few weeks ago, still waiting for response: https://bugs.launchpad.net/ubuntu/+source/docker-compose/+bug/1792824

**Comment by thaJeztah** [comment]
Looks like a fix for this was merged in the docker-credentials-helpers repository, but is not yet in the v0.6.1 release; https://github.com/docker/docker-credential-helpers/pull/29

**Comment by thaJeztah** [comment]
https://github.com/docker/docker-credential-helpers/pull/29 was merged, and is part of v0.6.2 of the credential helpers
